storage: TestReplicaBurstPendingCommandsAndRepropose is flaky

I've seen this on CircleCI, and reproed locally:

```
make stress STRESSFLAGS="-stderr -p 16 -maxfails 1" PKG=./storage 
TESTS=TestReplicaBurstPendingCommandsAndRepropose

--- FAIL: TestReplicaBurstPendingCommandsAndRepropose (0.01s)
	replica_test.go:6033: expected indexes [1 2 3 4 5 6 7 8 9 10], got [1 2 3 4 5 6 7 8 9 10 1 2 3 4 5 6 7 8 9 10]

2109 runs completed, 1 failures, over 33s
```

I've looked at the test a bit but didn't get very far. Ben, if you'd like me to investigate more, please assign back and I'll come with some questions.

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -5953,7 +5953,13 @@ func TestReplicaCancelRaftCommandProgress(t *testing.T) {
 func TestReplicaBurstPendingCommandsAndRepropose(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	var tc testContext
-	tc.Start(t)
+	sc := TestStoreContext()
+	// Ensure that the refresh call in this test is the only reason
+	// commands will get reproposed.
+	// TODO(bdarnell): why is this single-node test seeing a "new
+	// leader" (on rare occasions)? #8422
+	sc.TestingKnobs.DisableRefreshReasonNewLeader = true
+	tc.StartWithStoreContext(t, sc)
 	defer tc.Stop()
 
 	const num = 10

